core/src/Kokkos_View.hpp

```diff
@@ -1533,6 +1520,26 @@ class View : public ViewTraits<DataType, Properties...> {
     // Copy the input allocation properties with possibly defaulted properties
     alloc_prop prop_copy(arg_prop);
 
+    if (check_args == check_input_args::yes) {
+      size_t i0 = arg_layout.dimension[0];
+      size_t i1 = arg_layout.dimension[1];
+      size_t i2 = arg_layout.dimension[2];
+      size_t i3 = arg_layout.dimension[3];
+      size_t i4 = arg_layout.dimension[4];
+      size_t i5 = arg_layout.dimension[5];
+      size_t i6 = arg_layout.dimension[6];
+      size_t i7 = arg_layout.dimension[7];
+
+      const std::string& alloc_name =
+          static_cast<Kokkos::Impl::ViewCtorProp<void, std::string> const&>(
+              prop_copy)
+              .value;
+      Impl::runtime_check_rank(
+          traits::rank_dynamic,
+          std::is_same<typename traits::specialize, void>::value, i0, i1, i2,
+          i3, i4, i5, i6, i7, alloc_name.c_str());
```

Comment:
```suggestion
          i3, i4, i5, i6, i7, alloc_name);
```
Why converting to a C-string if the functions takes `std::string const&`?